server.js

```diff
@@ -1,33 +1,133 @@
-import express from 'express'
-import cors from 'cors'
-
-// If you're using one of our datasets, uncomment the appropriate import below
-// to get started!
-// 
-// import goldenGlobesData from './data/golden-globes.json'
-// import avocadoSalesData from './data/avocado-sales.json'
-// import booksData from './data/books.json'
-// import netflixData from './data/netflix-titles.json'
-// import topMusicData from './data/top-music.json'
-
-// Defines the port the app will run on. Defaults to 8080, but can be 
-// overridden when starting the server. For example:
-//
+import express from 'express';
+import cors from 'cors';
+import listEndpoints from 'express-list-endpoints';
+import topMusicData from './data/top-music.json';
+
+// Defines the port the app will run on. Defaults to 8080, but can // be overridden when starting the server. For example:
 //   PORT=9000 npm start
-const port = process.env.PORT || 8080
-const app = express()
+const port = process.env.PORT || 8080;
+const app = express();
 
 // Add middlewares to enable cors and json body parsing
-app.use(cors())
-app.use(express.json())
+app.use(cors());
+app.use(express.json());
 
-// Start defining your routes here
+// Default route
 app.get('/', (req, res) => {
-  res.send('Hello world')
-})
+  res.send(`List all possible endpoints by typing /endpoints on the URL bar`);
+});
+
+// Send list of all endpoints
+app.get('/endpoints', (req, res) => {
+  res.send(listEndpoints(app));
+});
+
+// Get all tracks. Optional: Filter by song title and/or artist.
+
+// Example: /tracks?artist=Ed Sheeran&songTitle=Perfect
+
+app.get('/tracks', (req, res) => {
+  const { songTitle, artist } = req.query;
+
+  let tracksToFilter = topMusicData;
+
+  if (songTitle) {
```

Comment:
Love this way of filtering, i didn't try it out myself but seems very handy to just do two ifs for the different searches on song/title and/or artist. Regarding the -1 i had to look up that because i didnt remember that index of actually returned -1 if the index wasnt found. Good knowledge to take away into next projects!